### PR TITLE
Python debugging lesson

### PR DESCRIPTION
**Pull request by jladan**
This is the lesson content for the "debugging with python" lesson.

### CODE REVIEW

File: lessons/python_debugging/lesson.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+To use `pdb`, you first need to import it, and add the line `pdb.set_trace()` where you'd like to start doing introspection. e.g.
+
+    import pdb
+
+    def average(xs):
+        pdb.set_trace()
+        length = len(xs)
+        acc = 0
+        for i in range(1, length)
```
Comment (QuLogic) [inline lessons/python_debugging/lesson.md]: Missing colon (I assume that's not the error.)

File: lessons/python_debugging/lesson.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ - **Lesson topic**: Introduction to Debugging in Python
+ - **Lesson content URL**: <https://github.com/UofTCoders/studyGroup/tree/gh-pages/lessons/python_debugging>
+
+## Preface: ##
+
+Debugging is really an art form. In this tutorial, we'll cover some concepts and tools you can use for debugging, but the skills need time to develop.
+
+The tutorial breaks debugging into three basic steps: testing code, finding bugs, and fixing them. Testing is a large topic on its own -- there's even a [blog/podcast about it](http://pythontesting.net/start-here/). We'll only cover some basics and the [`unittest`](https://docs.python.org/3.5/library/unittest.html) framework, so you may want to check out that blog to learn more.
+
+The bulk of this tutorial is about finding bugs. The only tool required are Python (`pdb` is included) and a text editor. However, I will give a demo using [Spyder](https://github.com/spyder-ide/spyder) to show how an IDE can help. Many other IDEs have the same (or more) features, but Spyder was convenient for me and is FOSS (free as in beer and speech).
```
Comment (QuLogic) [inline lessons/python_debugging/lesson.md]: tools

**Comment by jladan** [comment]
You're right, the missing colon was not the bug. those typos are fixed now.

**Comment by QuLogic** [comment]
Oh, just realized it's 6...

**Pull request merged**